src/GraphQL/Types/ArgumentInformation.cs

```diff
@@ -0,0 +1,179 @@
+        /// </summary>
+        protected ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, FieldType fieldType, TypeInformation typeInformation)
+            : this(parameterInfo, sourceType, fieldType, typeInformation, null)
+        {
+            if (parameterInfo.ParameterType == typeof(IResolveFieldContext))
+            {
+                Expression<Func<IResolveFieldContext, IResolveFieldContext>> expr = x => x;
+                Expression = expr;
+            }
+            else if (parameterInfo.ParameterType == typeof(CancellationToken))
+            {
+                Expression<Func<IResolveFieldContext, CancellationToken>> expr = x => x.CancellationToken;
+                Expression = expr;
+            }
+        }
+
+        /// <summary>
+        /// The method parameter.
+        /// </summary>
+        public ParameterInfo ParameterInfo { get; }
+
+        /// <summary>
+        /// The expected type of <see cref="IResolveFieldContext.Source"/>.
+        /// Should equal <c>TSourceType</c> within <see cref="AutoRegisteringObjectGraphType{TSourceType}"/>.
+        /// </summary>
+        public Type SourceType { get; }
+
+        /// <summary>
+        /// The <see cref="Types.FieldType"/> that the query argument will be added to.
+        /// </summary>
+        public FieldType FieldType { get; }
+
+        /// <summary>
+        /// The parsed type information of the method parameter.
+        /// </summary>
+        public TypeInformation TypeInformation { get; }
+
+        private LambdaExpression? _expression;
+        /// <summary>
+        /// Gets or sets a delegate in the form of a <see cref="LambdaExpression"/> to be used to populate
+        /// this method argument while building the field resolver.
+        /// <br/><br/>
+        /// If not set, a query argument will be added to the field and the argument's value will be used
+        /// to populate the method argument while building the field resolver.
+        /// <br/><br/>
+        /// The delegate must of the type
+        /// <see cref="Expression{TDelegate}">Expression</see>&lt;<see cref="Func{T, TResult}">Func</see>&lt;<see cref="IResolveFieldContext"/>, TReturnType&gt;&gt;
+        /// where TReturnType matches <see cref="ParameterInfo">ParameterInfo</see>.<see cref="ParameterInfo.ParameterType">ParameterType</see>.
+        /// </summary>
+        public LambdaExpression? Expression
+        {
+            get => _expression;
+            set
+            {
+                if (value != null && (value.ReturnType != ParameterInfo.ParameterType || value.Parameters.Count != 1 || value.Parameters[0].Type != typeof(IResolveFieldContext)))
+                {
+                    throw new ArgumentException($"Value must be a lambda expression delegate of type Func<IResolveFieldContext, {ParameterInfo.ParameterType.Name}>.");
+                }
+                _expression = value;
```

Comment:
So we can reset value?

Reply:
Yes (or clear it).  It's read/write, so long as it's valid.  A value of "null" is described in detail in the xml comments.